### ISSUE TITLE
Support multi-network for Pods (already exists in TF google provider)

### ISSUE THREAD

**Issue by anatanna** [terraform-google-modules/terraform-google-kubernetes-engine#2027]
### TL;DR

Support for **multi-network** feature is needed in module.
_enable_multi_networking_  argument is available in _google_container_cluster_ resource 
Description : (Optional) Whether multi-networking is enabled for this cluster.

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster
https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/container_cluster

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#additional_node_network_configs
https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/container_node_pool#additional_node_network_configs
```

### Detailed design

Multi-network feature is available in GKE as described here https://cloud.google.com/kubernetes-engine/docs/how-to/setup-multinetwork-support-for-pods.

According to gcloud cli the following configuration is available for node_pool 
  --additional-node-network network=NETWORK_NAME,subnetwork=SUBNET_NAME \
  --additional-pod-network subnetwork=subnet-dp,pod-ipv4-range=POD_IP_RANGE,max-pods-per-node=NUMBER_OF_PODS 

Which is implemented in terraform provider here for resource _container_node_pool_ https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#additional_node_network_configs 

So the possible design can be (under dedicated node_pool) as multiple blocks:
```markdown
  additional_pod_net_config {
     network = "NETWORK_NAME"
     subnetwork = "SUBNETWORK-NAME"
     }
  additional_pod_network_config {
     subnetwork = "SUBNETWORK-NAME"
     secondary_pod_range = POD_IP_RANGE
     max_pods_per_node=NUMBER_OF_PODS
     }
```

Please note that every additional_pod_network_config  can have its own max_pods_per_node that differs from node_pool.max_pods_per_node.

### Additional information
From Documentation 
To specify additional _node-network_ and _Pod-network_ interfaces, define the --additional-node-network and --additional-pod-network parameters multiple times as shown in the following example:
```markdown
--additional-node-network network=dataplane,subnetwork=subnet-dp \
--additional-pod-network subnetwork=subnet-dp,pod-ipv4-range=sec-range-blue,max-pods-per-node=8 \
--additional-pod-network subnetwork=subnet-dp,pod-ipv4-range=sec-range-green,max-pods-per-node=8 \
--additional-node-network network=managementdataplane,subnetwork=subnet-mp \
--additional-pod-network subnetwork=subnet-mp,pod-ipv4-range=sec-range-red,max-pods-per-node=4
```
_No response_

**Comment by lnrdll** [comment]
Is there a plan to add this feature to the module?
